### ISSUE TITLE
CNAME CAA records not fully returned

### ISSUE THREAD

**Issue by Kazzer** [silverwind/caa#5]
I'm noticing some odd behaviour with the handling of CAA records for alias targets where only one of the records is returned instead of all of the records.

Given this hypothetical configuration:

`alpha.domain.example`
CNAME: `bravo.example`
CAA: (unset)

`bravo.example`
A: `1.2.3.4`
CAA: `issue "amazon.com"`
CAA: `issue "comodoca.com"`
CAA: `issue "letsencrypt.org"`

Executing the following:

`await caa.matches('alpha.domain.example', 'letsencrypt.org');`

The library returns `false`, instead of `true`. Debugging a bit, executing:

`await caa('alpha.domain.example');`

returns `[{flags: 0, tag: 'issue', value: 'amazon.com', issuerCritical: false}]`.

I believe this section of code (https://github.com/silverwind/caa/blob/master/index.js#L126-L130) returns too eagerly and just returns after one match instead of filtering the list to all matches.

**Comment by Kazzer** [comment]
I believe an example fix for this would be:
```
@@ -123,11 +123,7 @@
   // If A(X) is not null, and CAA(A(X)) is not empty, then R(X) = CAA(A(X)), otherwise
   if (alias) {
     if (records.CAA && records.CAA.length) {
-      for (const record of records.CAA) {
-        if (record.name === alias && record.data) {
-          return [record.data];
-        }
-      }
+      return records.CAA.filter((record) => record.name === alias && record.data).map((record) => record.data);
     }
   }
```

Reference: #6

**Comment by silverwind** [comment]
Indeed, it's an oversight in the alias handling. Thanks for the PR. It's fixed in 5.0.4.
